### ISSUE TITLE
[BUG] Unable to open Database file on Windows 10

### ISSUE THREAD

**Issue by matthme** [holochain/holochain#3860]
**Describe the bug**
Genesis fails on Windows 10 with the following error:
```
Conductor returned an error while using a ConductorApi: GenesisFailed { errors: [(CellId(DnaHash(uhC0kBqyrNzeWnX1RU_fC5Jny4SAv5sw3Xff1TTyPtlJxtWXY9OcN), AgentPubKey(uhCAkp9zyG164j6029CS0hhQXsdu3PncFr5TkbfeN66UZv-lNu-HP)), DatabaseError(SqliteError(SqliteFailure(Error { code: CannotOpen, extended_code: 14 }, Some("unable to open database file: C:\\Users\\Matthias\\AppData\\Roaming\\holochain-launcher-electron\\0.300.0-alpha.1\\default\\holochain\\0.3.0-beta-dev.48\\dbs\\databases\\authored\\authored-uhC0kBqyrNzeWnX1RU_fC5Jny4SAv5sw3Xff1TTyPtlJxtWXY9OcN-uhCAkp9zyG164j6029CS0hhQXsdu3PncFr5TkbfeN66UZv-lNu-HP.sqlite3"))))), (CellId(DnaHash(uhC0kyMpkuK_eeMGIvT1IOqxEJfXCxg2ICwhADZfsMT3jRAwkJL6C), AgentPubKey(uhCAkp9zyG164j6029CS0hhQXsdu3PncFr5TkbfeN66UZv-lNu-HP)), DatabaseError(SqliteError(SqliteFailure(Error { code: CannotOpen, extended_code: 14 }, Some("unable to open database file: C:\\Users\\Matthias\\AppData\\Roaming\\holochain-launcher-electron\\0.300.0-alpha.1\\default\\holochain\\0.3.0-beta-dev.48\\dbs\\databases\\authored\\authored-uhC0kyMpkuK_eeMGIvT1IOqxEJfXCxg2ICwhADZfsMT3jRAwkJL6C-uhCAkp9zyG164j6029CS0hhQXsdu3PncFr5TkbfeN66UZv-lNu-HP.sqlite3")))))] }
```

**Expected behavior**
The database can be opened.

**System information:**
 - OS: Windows 10
 - Holochain version 0.3.0-beta-dev.48

**Steps to reproduce**
Download [this](https://github.com/holochain/launcher-electron/releases/tag/v0.300.0-alpha.2) launcher release and run it on Windows 10. 

**Additional context**
This issue started showing up somewhere in the 0.3.x line of holochain. I'm not quite sure where the problem is located but the same error shows up when running Moss 0.11.x in development mode on Windows and it has not shown up when Moss was still using holochain 0.2 (Moss 0.10.x) so something seems to have changed since then in holochain that affects this, even if it's an issue with file permissions.

**Comment by ThetaSinner** [comment]
I'm going to try and reproduce this on Windows when I next have access to a machine with Windows on (next few days).

I'll also look at changing the database config so that it is less likely to get corrupted in hte first place

**Comment by abe-njama** [comment]
@matthme this is issue is now resolved in both holochain `0.4.0-dev.13` and `0.3.2-rc.0` courtesy of https://github.com/holochain/holochain/pull/4070

While this improvement should help manage the db name in the path, the general recommendation is that the path name be kept short as Windows doesn't lengthy paths quite well.
